PhpStorm Protocol (Win)/run_editor.js

```diff
@@ -10,16 +10,41 @@ var settings = {
 };
 
 
-// don't change anything below this line, unless you're know what you're doing
-var url = WScript.Arguments(0),
-	match = /^pstorm:\/\/open\/\?url=file:\/\/(.+)&line=(\d+)$/.exec(url),
-	editor = '"c:\\' + (settings.x64 ? 'Program Files (x86)' : 'Program Files') + '\\JetBrains\\' + settings.folder_name + '\\bin\\PhpStorm.exe" --line %line% "%file%"';
+// don't change anything below this line, unless you know what you're doing
+var url = WScript.Arguments( 0 ),
+	match = /^pstorm:\/\/open\/\?url=file:\/\/(.+)&line=(\d+)$/.exec( url ),
+	project = '',
+	editor = '"C:\\' + (settings.x64 ? 'Program Files (x86)' : 'Program Files') + '\\JetBrains\\' + settings.folder_name + '\\bin\\PhpStorm.exe"';
 
 if ( match ) {
-	var	shell = new ActiveXObject('WScript.Shell'),
-		file = decodeURIComponent(match[1]).replace(/\+/g, ' '),
-		command = editor.replace(/%line%/g, match[2]).replace(/%file%/g, file).replace(/\//g, '\\')/*.replace(/\\/g, '\\\\')*/;
 
-	shell.Exec(command);
-	shell.AppActivate(settings.window_title);
+	var shell = new ActiveXObject( 'WScript.Shell' ),
+		fso = new ActiveXObject( 'Scripting.FileSystemObject' ),
+		file = decodeURIComponent( match[1] ).replace( /\+/g, ' ' ),
+		searchPath = file.replace( /\//g, '\\' );
+
+	while( searchPath.lastIndexOf( '\\' ) != -1 ) {
+		searchPath = searchPath.substring( 0, searchPath.lastIndexOf( '\\' ) );
+
+		if( fso.FileExists( searchPath+'\\.idea\\.name' ) ) {
+			project = searchPath;
+			break;
+		}
+	}
+
+	if( project != '' ) {
+		editor += ' "%project%"';
+	}
+
+	editor += ' --line %line% "%file%"';
+
+	var command = editor.replace( /%line%/g, match[2] )
+						.replace( /%file%/g, file )
+						.replace( /%project%/g, project )
+						.replace( /\//g, '\\' )
+						/*.replace(/\\/g, '\\\\')*/;

```

Comment:
Debug commented-out code doesn't really do anything. Please remove this line and line below it.